tensorzero-core/src/providers/fireworks/mod.rs

```diff
@@ -128,36 +128,50 @@ impl TryFrom<Credential> for FireworksCredentials {
 }
 
 impl FireworksCredentials {
+    /// Get API key, logging errors at the top level when the entire credential chain fails
     pub fn get_api_key<'a>(
         &'a self,
         dynamic_api_keys: &'a InferenceCredentials,
+    ) -> Result<&'a SecretString, Error> {
+        self.get_api_key_with_fallbacks(dynamic_api_keys)
+            .inspect_err(|err| {
+                err.log();
+            })
+    }
+
+    /// Recursive implementation that tries default credentials then falls back, without logging
+    fn get_api_key_with_fallbacks<'a>(
+        &'a self,
+        dynamic_api_keys: &'a InferenceCredentials,
     ) -> Result<&'a SecretString, Error> {
         match self {
             FireworksCredentials::Static(api_key) => Ok(api_key),
             FireworksCredentials::Dynamic(key_name) => {
                 dynamic_api_keys.get(key_name).ok_or_else(|| {
-                    ErrorDetails::ApiKeyMissing {
+                    Error::new_without_logging(ErrorDetails::ApiKeyMissing {
```

Comment:
can we change this to a `DelayedError` and then call `.suppress_logging_of_error_message()`? 

So like `DelayedError::new(ErrorDetails::...).suppress_logging_of_error_message()`.

We're trying to improve error patterns to a more expressive system and one of the goals is to remove `new_without_logging`.